rest_controllers/AuthRestController.php

```diff
@@ -31,25 +31,32 @@ public function authenticate($authPayload)
             return;
         }
         if (!empty($_SESSION['api']) && !empty($_SESSION['site_id'])) {
-            $encoded_api = bin2hex(trim($_SESSION['api']));
-            $encoded_site = bin2hex(trim($_SESSION['site_id']));
+            $encoded_api_site = bin2hex(trim($_SESSION['api']) . trim($_SESSION['site_id']));
         } else {
             http_response_code(401);
             return;
         }
 
+        // Use base64 (except for the special characters which are + and /) in Bearer tokens
+        // (note rfc6750 allows more special characters if wish to support in the future)
+        $new_token = produceRandomString(32, "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789");
+        if (empty($new_token)) {
+            http_response_code(500);
+            return;
+        }
+
         $user = sqlQuery("SELECT id FROM users_secure WHERE username = ?", array($authPayload['username']));
 
         $sql = " INSERT INTO api_token SET";
         $sql .= "     user_id=?,";
-        $sql .= "     token=(SELECT LEFT(SHA2(CONCAT(NOW(), RAND(), UUID()), 512), 32)),";
+        $sql .= "     token=?,";
         $sql .= "     expiry=DATE_ADD(NOW(), INTERVAL 1 HOUR)";
 
-        sqlInsert($sql, array($user["id"]));
+        sqlInsert($sql, [$user["id"], $new_token]);
 
         $token = sqlQuery("SELECT token FROM api_token WHERE user_id = ? ORDER BY id DESC", array($user["id"]));
```

Comment:
The only reason this fetch was ran is because the mysql random and uuid generators were used to generate the encoded token. If not using these routines then no reason to requery api table. I suppose it won't hurt to leave.

Reply:
Thanks for pointing that out. Removed the query.